buildSrc/src/main/kotlin/Config.kt

```diff
@@ -15,6 +15,51 @@
  * limitations under the License.
  */
 
+
+/**
+ * Enum describing operating systems we can build on.
+ *
+ * We need to track this in order to control which Kotlin Multiplatform tasks we can safely
+ * create on the given Host OS.
+ */
+enum class OperatingSystem {
```

Comment:
Nice infrastructure. Would be great if we could find a cleaner way (than the current if-else-statements) to add all the targets/tasks based on this infrastructure. But at least building blocks for future clean ups 😅